lib/std/fs/Dir.zig

```diff
@@ -377,6 +420,16 @@ pub const Iterator = switch (builtin.os.tag) {
                     linux.DT.LNK => .sym_link,
                     linux.DT.REG => .file,
                     linux.DT.SOCK => .unix_domain_socket,
+                    linux.DT.UNKNOWN => query_kind: {
+                        if (!self.opts.query_kind) {
+                            break :query_kind .unknown;
+                        }
+                        if (self.dir.statFile(name)) |info| {
```

Comment:
I'm not into zig, but `statFile()` seems to resolve to `posix.fstatat(self.fd, sub_path, 0)` , so no `AT_SYMLINK_NOFOLLOW`. Following symbolic links here doesn't match the semantic of `d_type` from `getdents64()` and might cause Walker to walk in a loop.